jerry-core/ecma/operations/ecma-array-object.c

```diff
@@ -120,44 +159,208 @@ ecma_op_array_is_fast_array (ecma_extended_object_t *array_p) /**< ecma-array-ob
   return array_p->u.array.u.length_prop & ECMA_FAST_ARRAY_FLAG;
 } /* ecma_op_array_is_fast_array */
 
+/**
+ * Allocate a new fast access mode array object with 0 length
+ *
+ * @return pointer to the constructed fast access mode array object
+ */
+inline ecma_object_t * JERRY_ATTR_ALWAYS_INLINE
+ecma_op_new_empty_array (void)
```

Comment:
Do we need this if we have the method below? We could directly call the one below, using an assert instead of a out-of-memory exit does not change much.

Reply:
Good idea!